src/azure-cli/azure/cli/command_modules/rdbms/flexible_server_custom_mysql.py

```diff
@@ -0,0 +1,483 @@
+        tags=tags)
+
+    if assign_identity:
+        parameters.identity = mysql_flexibleservers.Identity(
+            type=mysql_flexibleservers.models.ResourceIdentityType.system_assigned.value)
+
+    return resolve_poller(
+        server_client.create(resource_group_name, server_name, parameters), cmd.cli_ctx,
+        '{} Server Create'.format(logging_name))
+
+
+def flexible_server_connection_string(
+            server_name='{server}', database_name='{database}', administrator_login='{login}',
+            administrator_login_password='{password}'):
+    host = '{}.mysql.database.azure.com'.format(server_name)
+    return {
+        'connectionStrings': _create_mysql_connection_strings(host, administrator_login, administrator_login_password, database_name)
+    }
+
+
+def _create_mysql_connection_strings(host, user, password, database):
+    result = {
+        'mysql_cmd': "mysql {database} --host {host} --user {user} --password={password}",
+        'ado.net': "Server={host}; Port=3306; Database={database}; Uid={user}; Pwd={password};",
+        'jdbc': "jdbc:mysql://{host}:3306/{database}?user={user}&password={password}",
+        'jdbc Spring': "spring.datasource.url=jdbc:mysql://{host}:3306/{database}  "
+                       "spring.datasource.username={user}  "
+                       "spring.datasource.password={password}",
+        'node.js': "var conn = mysql.createConnection({{host: '{host}', user: '{user}', "
+                   "password: {password}, database: {database}, port: 3306}});",
+        'php': "host={host} port=3306 dbname={database} user={user} password={password}",
+        'python': "cnx = mysql.connector.connect(user='{user}', password='{password}', host='{host}', "
+                  "port=3306, database='{database}')",
+        'ruby': "client = Mysql2::Client.new(username: '{user}', password: '{password}', "
+                "database: '{database}', host: '{host}', port: 3306)",
+    }
+
+    connection_kwargs = {
+        'host': host,
+        'user': user,
+        'password': password if password is not None else '{password}',
+        'database': database
+    }
+
+    for k, v in result.items():
+        result[k] = v.format(**connection_kwargs)
+    return result
+
+
+def _form_response(username, sku, location, id, host, version, password, connection_string, database_name, firewall_id=None, subnet_id=None):
+    output = {
+        'host': host,
+        'username': username,
+        'password': password,
+        'skuname': sku,
+        'location': location,
+        'id': id,
+        'version': version,
+        'databaseName': database_name,
+        'connectionString': connection_string
+    }
+    if firewall_id is not None:
+        output['firewallName'] = firewall_id
+    if subnet_id is not None:
+        output['subnetId'] = subnet_id
+    return output
+
+
+def _update_local_contexts(cmd, server_name, resource_group_name, location, user):
```

Comment:
Why need to update local context value manually ? For local context feature, we only need to define the local_context_attribute for the parameter and the framework will do the read/write work automatically.

You can reference this document: https://github.com/Azure/azure-cli/blob/dev/doc/local_context_guidelines.md

Reply:
We automatically generate some arguments without users' input (e.g., server name, username etc). In this case, we weren't able to save the local context values just by defining the attribute. Is there another way we can set it?